resources/migrations/001_update_migrations.yaml

```diff
@@ -6715,6 +6715,18 @@ databaseChangeLog:
             tableName: metabase_field
             constraintName: idx_unique_field
 
+  - changeSet:
+      id: v49.2024-06-27T00:00:09
+      author: calherries
+      comment: Set is_defective_duplicate=TRUE fields that shouldn't exist to have active=FALSE
+      changes:
+        - sql:
+            sql: >-
+              UPDATE metabase_field
+              SET active = false
+              WHERE is_defective_duplicate AND (nfc_path = concat('["', name, '"]') OR nfc_path IS NULL);
```

Comment:
We soft-delete two categories of defective duplicate fields:
- `nfc_path IS NULL`: that means the field is not nested, and there's at least one other non-nested field with the same name. I'm not sure this is possible in the current app but there could have been a bug in the past that allowed this in MySQL and H2.
- `(nfc_path = concat('["', name, '"]')`: this is the exact scenario that was caused by https://github.com/metabase/metabase/pull/43812 and fixed by https://github.com/metabase/metabase/pull/44465. These columns can be replaced because using them in a query currently can only produce the same results as using the corresponding non-defective field. See the "Results" section in [this comment](https://github.com/metabase/metabase/issues/44659#issuecomment-2192902017) for more on this.

Reply:
I'm deliberately not soft-deleting all defective duplicate fields, because it's not clear they should actually be deactivated. 

For example, it was possible with MySQL and H2 for a nested JSON field to have the same name as a non-nested column, or a field with a different `nfc_path`. This was not possible with Postgres before, because it would violate the `idx_uniq_field_table_id_parent_id_name_2col` constraint.

I intend to expand the current `idx_unique_field` constraint to allow these in a following PR.

Reply:
yea I agree that we should be conservative here and only deactivate the scenario we know about.